### PR TITLE
Jasmine

### PR DESCRIPTION
**Pull request by Jasmine-129**
...

### CODE REVIEW

**Review by archwu — APPROVED** [review]

File: src/ui/guiforcase4/AddVehicles.java [hunk 1]
```diff
@@ -65,51 +83,135 @@ private void addComponents(JFrame frame, JPanel panel) {
...
+    public String[] getMakesFromDatabase() {
+        ArrayList<Vehicle> vehicles = connect.executeVehicleQuery();
+        String[] makes = new String[vehicles.size()];
+        for (int i = 0; i < vehicles.size(); i++) {
+            makes[i] = vehicles.get(i).getMake();
+        }
+        return makes;
+    }
+
+    public String[] removeDuplicates(String[] array) {
```
Comment (kumarlakshya24) [inline src/ui/guiforcase4/AddVehicles.java:190]: The idea is good, but I was wondering if there is any way to check for duplicates within the SQL query itself?

**Review by kumarlakshya24 — COMMENTED** [review]
Leaving a comment to verify if duplicates can be eliminated using SQL query itself otherwise currently being handled by the removeDuplicates method

File: src/ui/guiforcase4/AddVehicles.java (hunk 1 above)
Comment (archwu) [reply src/ui/guiforcase4/AddVehicles.java:190]: We will be refactoring this part in upcoming version

Comment (Jasmine-129) [reply src/ui/guiforcase4/AddVehicles.java:190]: You are right, Lakshya! Of Course, we can use SQL query to fetch distinct values of attributes. This query "SELECT DISTINCT column1, column2, column3" can only get distinct values of column1, I was too lazy to separate them into three methods, so...    I will try to improve them later. Thank you~

**Pull request merged**